returnn/tf/layers/basic.py

```diff
@@ -2059,6 +2060,8 @@ def get_out_data_from_opts(cls, name, sources, axis, dtype="int32", keepdims=Tru
         out.feature_dim_axis = None
       out.size_placeholder = {0: size for (i, size) in source.size_placeholder.items() if i == axis_wo_b}
     out.sparse = sparse
+    if out.feature_dim_axis is None:
+      out.dim = None
```

Comment:
This is wrong. `dim` specifies the number of classes in case it is sparse (i.e. when it does not have a feature dim).
```suggestion
    if out.feature_dim_axis is None and not out.sparse:
      out.dim = None
```

Reply:
Oops, thanks! I added your suggestion.